java/src/main/native/src/TableJni.cpp

```diff
@@ -1647,7 +1647,8 @@ Java_ai_rapids_cudf_Table_readAndInferJSONFromDataSource(JNIEnv* env,
         .normalize_whitespace(static_cast<bool>(normalize_whitespace))
         .mixed_types_as_string(mixed_types_as_string)
         .strict_validation(strict_validation)
-        .keep_quotes(keep_quotes);
+        .keep_quotes(keep_quotes)
+        .prune_columns(false);
```

Comment:
I am surprised we need to pass an option here. What happens if it's set to true? It might be worth a comment...

Reply:
I was just being defensive. This code is trying to infer the schema. I don't remember exactly what happens if I set it to true, but I think it ends up pruning everything.

Reply:
> ends up pruning everything

:)